event-hubs.api.md

```diff
@@ -0,0 +1,287 @@
+## API Report File for "@azure/event-hubs"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AbortSignalLike } from '@azure/abort-controller';
+import { DataTransformer } from '@azure/core-amqp';
+import { DefaultDataTransformer } from '@azure/core-amqp';
+import { MessagingError } from '@azure/core-amqp';
+import { RetryOptions } from '@azure/core-amqp';
+import { Span } from '@azure/core-tracing';
+import { SpanContext } from '@azure/core-tracing';
+import { SpanOptions } from '@azure/core-tracing';
+import { TokenCredential } from '@azure/core-amqp';
+import { TokenType } from '@azure/core-amqp';
+import { WebSocketImpl } from 'rhea-promise';
+
+// @public
+export interface AbortSignalOptions {
+    abortSignal?: AbortSignalLike;
+}
+
+// @public
+export interface Checkpoint extends PartitionContext {
+    offset: number;
+    sequenceNumber: number;
+}
+
+// @public
+export enum CloseReason {
+    OwnershipLost = "OwnershipLost",
+    Shutdown = "Shutdown"
+}
+
+// @public
+export interface CreateBatchOptions {
+    abortSignal?: AbortSignalLike;
+    maxSizeInBytes?: number;
+    partitionId?: string;
+    partitionKey?: string;
+}
+
+export { DataTransformer }
+
+export { DefaultDataTransformer }
+
+// @public
+export interface EventData {
+    body: any;
+    properties?: {
+        [key: string]: any;
+    };
+}
+
+// @public
+export interface EventDataBatch {
+    batchMessage: Buffer | undefined;
+    count: number;
+    // @internal
+    _messageSpanContexts: SpanContext[];
+    partitionId?: string;
+    partitionKey?: string;
+    sizeInBytes: number;
+    tryAdd(eventData: EventData, options?: TryAddOptions): boolean;
```

Comment:
TryAddOptions? Why does this exist? It only has a parent context. 🤔

Reply:
Convention for optional parameters. 

Although...perhaps worth discussion if there are criteria where we just promote it to an argument....

Reply:
Idiomatic, we're going to leave it as is.